src/components/JSONSnippetImportDialog.tsx

```diff
@@ -0,0 +1,204 @@
+import React, { useState } from "react"
+import {
+  Dialog,
+  DialogContent,
+  DialogHeader,
+  DialogTitle,
+  DialogFooter,
+} from "@/components/ui/dialog"
+import { Button } from "@/components/ui/button"
+import { Input } from "@/components/ui/input"
+import { useAxios } from "@/hooks/use-axios"
+import { useToast } from "@/hooks/use-toast"
+import { useLocation } from "wouter"
+import { useGlobalStore } from "@/hooks/use-global-store"
+import { convertCircuitJsonToTscircuit } from "circuit-json-to-tscircuit"
+
+interface JSONSnippetImportDialogProps {
+  open: boolean
+  onOpenChange: (open: boolean) => void
+}
+
+const isValidURL = (url: string) => {
+  try {
+    new URL(url)
+    return true
+  } catch {
+    return false
+  }
+}
+
+export function JSONSnippetImportDialog({
+  open,
+  onOpenChange,
+}: JSONSnippetImportDialogProps) {
+  const [jsonUri, setJsonUri] = useState("")
+  const [file, setFile] = useState<File | null>(null)
+  const [isLoading, setIsLoading] = useState(false)
+  const [error, setError] = useState<string | null>(null)
+  const axios = useAxios()
+  const { toast } = useToast()
+  const [, navigate] = useLocation()
+  const isLoggedIn = useGlobalStore((s) => Boolean(s.session))
+  const session = useGlobalStore((s) => s.session)
+
+  const handleFileChange = async (e: React.ChangeEvent<HTMLInputElement>) => {
+    const selectedFile = e.target.files?.[0]
+    if (selectedFile && selectedFile.type === "application/json") {
+      setFile(selectedFile)
+    } else {
+      setError("Please select a valid JSON file.")
+    }
+  }
+
+  const handleImport = async () => {
+    let jsonData
+
+    if (file) {
+      try {
+        const fileText = await file.text()
+        jsonData = JSON.parse(fileText)
+      } catch (err) {
+        setError("Error reading JSON file. Please ensure it is valid.")
+        return
+      }
+    } else if (isValidURL(jsonUri)) {
+      setIsLoading(true)
+      setError(null)
+
+      try {
+        jsonData = (
+          await axios.get(jsonUri, {
```

Comment:
the variable names are really not clear here and you should not need to call the API

But the variable names are really not good, note that the word "data" is literally a banned word https://github.com/tscircuit/handbook/blob/main/guides/code.md